### PR TITLE
[krakenfutures] fetchFundingRate(s)

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

File: ts/src/krakenfutures.ts [hunk 1]
```diff
@@ -1591,6 +1591,90 @@ export default class krakenfutures extends Exchange {
...
+    async fetchFundingRate (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const fundingRates = await this.fetchFundingRates ([ symbol ], params);
+        for (let i = 0; i < fundingRates.length; i++) {
+            const entry = fundingRates[i];
+            if (entry['symbol'] === market['symbol']) {
+                return entry;
+            }
+        }
```
Comment (carlosmiei) [inline ts/src/krakenfutures.ts:1603]: @youknowone `fetchFundingRates` should return the response already filtered by `symbol`, isn't this enough? `const response = this.safeValue(fundingRates,0)` ?

Comment (youknowone) [reply ts/src/krakenfutures.ts]: Thanks, I updated it. I realized fetchFundingRate in base Exchange has default implementation. Should I remove this one and let it use the default implementation?

File: ts/src/krakenfutures.ts [hunk 2]
```diff
@@ -1591,6 +1591,86 @@ export default class krakenfutures extends Exchange {
...
+            const entry = tickers[i];
+            const entry_symbol = this.safeValue (entry, 'symbol');
+            if (marketIds !== undefined) {
+                if (!this.inArray (entry_symbol, marketIds)) {
+                    continue;
+                }
+            }
+            const market = this.safeMarket (entry_symbol);
+            const parsed = this.parseFundingRate (entry, market);
+            fundingRates.push (parsed);
```
Comment (carlosmiei) [inline ts/src/krakenfutures.ts:1619]: @youknowone one more thing, I think this method should return an object indexed by the symbol and not a list

Comment (youknowone) [reply ts/src/krakenfutures.ts:1619]: Ah, I finally realized `filter_by_array` can return either array or object.
I will add index_by call to the result.

**Comment by youknowone** [comment]
Thank you so much for the review. I applied the review.

File: ts/src/krakenfutures.ts (hunk 2 above)
Comment (carlosmiei) [reply ts/src/krakenfutures.ts:1610]: @youknowone thanks!

**Comment by carlosmiei** [comment]
LGTM ✅ 
```
p krakenfutures fetchFundingRates '["BTC/USD:USD"]'
Python v3.10.9
CCXT v3.1.34
krakenfutures.fetchFundingRates(['BTC/USD:USD'])
{'BTC/USD:USD': {'datetime': '2023-06-13T08:19:28.091Z',
                 'estimatedSettlePrice': None,
                 'fundingDatetime': None,
                 'fundingRate': 0.5403157744358391,
                 'fundingTimestamp': None,
                 'indexPrice': None,
                 'info': {'ask': '26106',
                          'askSize': '0.5068',
                          'bid': '26100',
                          'bidSize': '0.397',
                          'fundingRate': '0.540315774435839127',
                          'fundingRatePrediction': '0.437391010194719323',
                          'indexPrice': '26092.71',
                          'last': '26105',
                          'lastSize': '0.0001',
                          'lastTime': '2023-06-13T08:19:28.091Z',
                          'markPrice': '26103',
                          'open24h': '25866',
                          'openInterest': '1365.19690000000',
                          'pair': 'XBT:USD',
                          'postOnly': False,
                          'suspended': False,
                          'symbol': 'pf_xbtusd',
                          'tag': 'perpetual',
                          'vol24h': '2194.0098',
                          'volumeQuote': '56910792.672'},
                 'interestRate': None,
                 'markPrice': None,
                 'nextFundingDatetime': None,
                 'nextFundingRate': 0.4373910101947193,
                 'nextFundingTimestamp': None,
                 'previousFundingDatetime': None,
                 'previousFundingRate': None,
                 'previousFundingTimestamp': None,
                 'symbol': 'BTC/USD:USD',
                 'timestamp': 1686644368091}}
```

File: ts/src/krakenfutures.ts [hunk 3]
```diff
@@ -1645,12 +1645,13 @@ export default class krakenfutures extends Exchange {
...
         //
         const marketId = this.safeString (ticker, 'symbol');
+        const symbol = this.symbol (marketId);
         const timestamp = this.parse8601 (this.safeString (ticker, 'lastTime'));
         const fundingRate = this.safeNumber (ticker, 'fundingRate');
         const nextFundingRate = this.safeNumber (ticker, 'fundingRatePrediction');
         return {
             'info': ticker,
-            'symbol': marketId,
+            'symbol': symbol,
```
Comment (youknowone) [inline ts/src/krakenfutures.ts:1654]: I originally wrote it but it wasn't compatible with base exchange's fetchFundingRates implementation

Comment (carlosmiei) [reply ts/src/krakenfutures.ts:1654]: @youknowone I tested it and seems to be working fine
```
 p krakenfutures fetchFundingRate "BTC/USD:USD"   
Python v3.10.9
CCXT v3.1.34
krakenfutures.fetchFundingRate(BTC/USD:USD)
{'datetime': '2023-06-13T09:03:36.454Z',
 'estimatedSettlePrice': None,
 'fundingDatetime': None,
 'fundingRate': 0.48863570971999126,
 'fundingTimestamp': None,
 'indexPrice': None,
 'info': {'ask': '26210',
          'askSize': '0.395',
          'bid': '26206',
          'bidSize': '0.5037',
          'fundingRate': '0.488635709719991274',
          'fundingRatePrediction': '0.574588639840842062',
          'indexPrice': '26193.62',
          'last': '26205',
          'lastSize': '0.0001',
          'lastTime': '2023-06-13T09:03:36.454Z',
          'markPrice': '26208',
          'open24h': '25953',
          'openInterest': '1370.38900000000',
          'pair': 'XBT:USD',
          'postOnly': False,
          'suspended': False,
          'symbol': 'pf_xbtusd',
          'tag': 'perpetual',
          'vol24h': '2228.6211',
          'volumeQuote': '57829701.9162'},
 'interestRate': None,
 'markPrice': None,
 'nextFundingDatetime': None,
 'nextFundingRate': 0.5745886398408421,
 'nextFundingTimestamp': None,
 'previousFundingDatetime': None,
 'previousFundingRate': None,
 'previousFundingTimestamp': None,
 'symbol': 'BTC/USD:USD',
 'timestamp': 1686647016454}
```

**Pull request merged**